pkgdb2client/admin.py

```diff
@@ -454,10 +446,39 @@ def __handle_request_unretire(actionid, action):
     action['info']['pkg_namespace'] = action['package'].get(
         'namespace', 'rpms')
 
-    msgs = utils.check_package_creation(
-        action['info'], bugid, PKGDBCLIENT, action['user'])
+    if action['info']['pkg_review_url']:
+        bugid = utils.get_bug_id_from_url(action['info']['pkg_review_url'])
+        msgs = utils.check_package_creation(
+            action['info'], bugid, PKGDBCLIENT, action['user'])
+    else:
+        msgs = utils.check_branch_creation(
+            PKGDBCLIENT, action['package']['name'],
+            action['collection']['branchname'], action['user'], new_pkg=True,
+            namespace=action['info']['pkg_namespace'])
+
+        package = PKGDBCLIENT.get_package(
+            action['package']['name'],
+            branches=[action['info']['pkg_collection']],
+            namespace=action['info']['pkg_namespace'],
+        )["packages"][0]
+        rawhide_package = PKGDBCLIENT.get_package(
+            action['package']['name'], branches=["master"],
+            namespace=action['info']['pkg_namespace'],
+        )["packages"][0]
+
+        last_pkg_change = datetime.datetime.fromtimestamp(
+            package["status_change"])
+        now = datetime.datetime.now()
+        delta = now - last_pkg_change
+        if delta.days < 14:
+            msgs["good"].append("Package was retired less than 14 days ago")
+        elif rawhide_package.status != "Retired":
+            msgs["good"].append("Package is not retired in Rawhide")
```

Comment:
This is for unretirement requests for branches other than rawhide, because there should be no unretire requests for Rawhide if Rawhide is not retired. If Rawhide is not retired all other possible retired branches can be unretired without a re-review. Therefore the unretiretment request is good, if the package is not retired in Rawhide.